### PR TITLE
feat: Implement getSites rest endpoint and services in order to retrieve sites with navigations - EXO-65477 - Meeds-io/MIPs#72

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: component/portal/src/main/java/org/exoplatform/portal/mop/rest/NavigationRest.java [hunk 1]
```diff
@@ -202,6 +204,24 @@ public Response getNavigationCategories() {
     }
   }
 
+
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/rest/NavigationRest.java]: we need a dedicated rest service for sites

File: component/api/src/main/java/org/exoplatform/portal/mop/storage/SiteStorage.java [hunk 2]
```diff
@@ -64,4 +64,6 @@ public interface SiteStorage extends ModelDataStorage {
 
   Container getSharedLayout(String siteName);
 
+  List<PortalData> getPortalSitesOrderedByDisplayOrder();
```
Comment (azayati) [inline component/api/src/main/java/org/exoplatform/portal/mop/storage/SiteStorage.java]: this should be generic

File: component/portal/src/main/java/org/exoplatform/portal/jdbc/entity/SiteEntity.java [hunk 3]
```diff
@@ -68,6 +68,14 @@
         + " WHERE s.siteType = :siteType"
         + " ORDER BY s.name ASC"
 )
+
+@NamedQuery(
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/jdbc/entity/SiteEntity.java]: this should be generic

File: component/portal/src/main/java/org/exoplatform/portal/mop/dao/SiteDAO.java [hunk 4]
```diff
@@ -42,4 +42,6 @@ public interface SiteDAO extends GenericDAO<SiteEntity, Long> {
 
   List<String> findUserSites(int offset, int limit);
 
+  List<SiteEntity> findAPortalSitesOrderedByDisplayOrder();
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/dao/SiteDAO.java]: this should be generic

**Review by azayati — CHANGES_REQUESTED** [review]

File: component/portal/src/main/java/org/exoplatform/portal/jdbc/entity/SiteEntity.java [hunk 5]
```diff
@@ -68,6 +68,8 @@
         + " WHERE s.siteType = :siteType"
         + " ORDER BY s.name ASC"
 )
+
+
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/jdbc/entity/SiteEntity.java]: remove that change

File: component/portal/src/main/java/org/exoplatform/portal/mop/rest/NavigationRest.java [hunk 6]
```diff
@@ -16,6 +16,7 @@
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
+import org.exoplatform.portal.mop.rest.model.UserNodeRestEntity;
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/rest/NavigationRest.java:19]: Remove that change

File: component/portal/src/main/java/org/exoplatform/portal/mop/rest/SiteRestService.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+  private static final Log LOG = ExoLogger.getLogger(SiteRestService.class);
+
+  private LayoutService    layoutService;
+
+  public SiteRestService(LayoutService layoutService) {
+    this.layoutService = layoutService;
+  }
+
+    @GET
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/rest/SiteRestService.java]: indentation

File: component/portal/src/main/java/org/exoplatform/portal/mop/rest/SiteRestService.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    @Produces(MediaType.APPLICATION_JSON)
+    @RolesAllowed("users")
+    @Operation(summary = "Gets navigations categories for UI", description = "Gets navigations categories for UI", method = "GET")
+    @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+            @ApiResponse(responseCode = "500", description = "Internal server error"), })
+    public Response getSitesWithNavigationNodes(@Context HttpServletRequest request,
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/rest/SiteRestService.java]:
```suggestion
    public Response getSites(@Context HttpServletRequest request,
```

File: component/portal/src/test/java/org/exoplatform/portal/mop/rest/NavigationRestTest.java [hunk 9]
```diff
@@ -36,6 +36,7 @@
 import org.exoplatform.portal.config.model.Page;
 import org.exoplatform.portal.mop.navigation.Scope;
 import org.exoplatform.portal.mop.page.PageKey;
+import org.exoplatform.portal.mop.rest.model.UserNodeRestEntity;
```
Comment (azayati) [inline component/portal/src/test/java/org/exoplatform/portal/mop/rest/NavigationRestTest.java]: Remove that change

**Review by azayati — CHANGES_REQUESTED** [review]

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**